pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/AggregationOperator.java

```diff
@@ -46,23 +45,23 @@
    * Constructor for the class.
    *
    * @param aggregationsInfoList List of AggregationInfo (contains context for applying aggregation functions).
+   * @param executor Aggregation Executor
    * @param projectionOperator Projection operator.
    * @param numTotalRawDocs Number of total raw documents.
    */
-  public AggregationOperator(List<AggregationInfo> aggregationsInfoList, MProjectionOperator projectionOperator,
-      long numTotalRawDocs) {
+  public AggregationOperator(List<AggregationInfo> aggregationsInfoList, AggregationExecutor executor,
```

Comment:
This moves operator wiring logic to the planner. Given that we have interface-impl pattern for AggregationExecutor, decision to instantiate executors will be based on segment metadata or brokerRequest. That decision should be made by planner. Operators then have single responsibility to to perform specific operation using the inputs.